pkg/filesystem/local_directory_linux.go

```diff
@@ -34,3 +43,32 @@ func (d *localDirectory) Mknod(name path.Component, perm os.FileMode, deviceNumb
 func clonefileImpl(oldFD int, oldName string, newFD int, newName string) error {
 	return status.Error(codes.Unimplemented, "Clonefile is only supported on Darwin")
 }
+
+// Uses `unmount` on relative pathnames with `fchdir`.
+func fchdir_unmountat(dfd int, mountname string) error {
+	fchdirLock.Lock()
+	defer fchdirLock.Unlock()
+
+	store, err := os.Getwd()
+	if err != nil {
+		return util.StatusWrap(err, "Could not store the working directory %#v.")
+	}
+	defer os.Chdir(store)
+
+	err = syscall.Fchdir(dfd)
+	if err != nil {
+		err = &fs.PathError{Op: "fchdir", Path: fmt.Sprintf("%d", dfd), Err: err}
```

Comment:
Don't bother wrapping the error with a PathError. Also including the file descriptor number in the error message is a bit too low-level and also inconsistent with the other methods. This is sufficient:
```
return util.StatusWrap(err, "Could not change working directory to parent of mount")
```